File: engines/java/src/main/java/io/github/davecookvectorlabs/windowcleaning/CleaningFrequency.java
package io.github.davecookvectorlabs.windowcleaning;

/**
 * Cleaning frequency options with volume discount factors.
 * Regular schedules reduce per-visit cost through route efficiency
 * and retained familiarity with the building.
 */
public enum CleaningFrequency {
    ONE_TIME(1.0),
    QUARTERLY(0.90),
    MONTHLY(0.80),
    WEEKLY(0.70);

    private final double discount;

    CleaningFrequency(double discount) {
        this.discount = discount;
    }

    public double getDiscount() {
        return discount;
    }
}

File: engines/php/src/CleaningFrequency.php
<?php

declare(strict_types=1);

namespace WindowCleaning;

enum CleaningFrequency: string
{
    case OneTime = 'one_time';
    case Quarterly = 'quarterly';
    case Monthly = 'monthly';
    case Weekly = 'weekly';

    public function discount(): float
    {
        return match ($this) {
            self::OneTime => 1.0,
            self::Quarterly => 0.90,
            self::Monthly => 0.80,
            self::Weekly => 0.70,
        };
    }
}
